builds/e2e/templates/e2e-run.yaml

```diff
@@ -0,0 +1,54 @@
+steps:
+- pwsh: |
+    $testFile = '$(binDir)/Microsoft.Azure.Devices.Edge.Test.dll'
+
+    if ($IsWindows)
+    {
+      # The windows installer script doesn't update the PATH in the current
+      # terminal session, so add `iotedge` and `docker` CLIs manually
+      $env:Path =
+        $env:Path,
+        $(Join-Path $env:ProgramFiles 'iotedge'),
+        $(Join-Path $env:ProgramFiles 'iotedge-moby') -join ';'
+
+      dotnet vstest $testFile --logger:trx
+    }
+    else
+    {
+      sudo --preserve-env dotnet vstest $testFile --logger:trx
+    }
+  displayName: Run tests
+  env:
+    E2E_EVENT_HUB_ENDPOINT: $(TestEventHubCompatibleEndpoint)
+    E2E_IOT_HUB_CONNECTION_STRING: $(TestIotHubConnectionString)
+    E2E_REGISTRIES__0__PASSWORD: $(TestContainerRegistryPassword)
+    E2E_ROOT_CA_PASSWORD: $(TestRootCaPassword)
+
+- task: PublishTestResults@2
+  displayName: Publish test results
+  inputs:
+    testResultsFormat: vstest
+    testResultsFiles: '**/*.trx'
+    searchFolder: $(Build.SourcesDirectory)/TestResults
+    testRunTitle: End-to-end tests ($(Build.BuildNumber) $(os) $(arch))
+    buildPlatform: $(arch)
+  condition: succeededOrFailed()
+
+- pwsh: |
+    $logDir = '$(Build.ArtifactStagingDirectory)/logs'
```

Comment:
just to confirm, $(Build.ArtifactStagingDirectory) will be flushed every time before build, right?

Reply:
Yes, the [docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#build-variables) for `Build.ArtifactStagingDirectory` say:

> This directory is purged before each new build, so you don't have to clean it up yourself.